cmd/compose/port.go

```diff
@@ -62,11 +68,40 @@ func portCommand(p *ProjectOptions, dockerCli command.Cli, backend api.Service)
 	return cmd
 }
 
+func printPorts(dockerCli command.Cli, publishers api.PortPublishers, opts portOptions) {
+	for _, publisher := range publishers {
+		if publisher.Protocol != opts.protocol {
+			continue
+		}
+
+		var format string
+		if strings.Contains(publisher.URL, ":") {
```

Comment:
would make sense to me `PortPublisher` has a `String()` method to handle this, aligned with docker/cli